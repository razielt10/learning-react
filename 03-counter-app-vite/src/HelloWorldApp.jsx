import React from 'react'

export const HelloWorldApp = () => {
  return (
    <div>
        React con Vite!!!
    </div>
  )
}

