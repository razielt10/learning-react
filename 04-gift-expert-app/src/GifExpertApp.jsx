import { useState } from "react"
import { AddCategory, GifGrid } from "./components"

export const GifExperApp = () => {

    const [categories, setCategories] = useState([])

    const onAddCategory = (newCat) => {
        // setCounter(counter + 1)
        // setCategories( [...categories, 'Otro'] )
        const found = categories.filter( c => {
            return c.toLowerCase() == newCat.toLowerCase()
        })
        if (found.length != 0) {
            console.log('ya tengo esa categoria')
            return 
        }

        setCategories( (c) =>  [...c, newCat] )
    } 

    return (
        <>
            {/* Titulo */}
            <h1>GifExpertApp</h1>
            
            {/* input */}
            <AddCategory onAddCategory={ onAddCategory } />

            {/* Listado Gif */}
            {
                categories.map( (cat) => (
                    <GifGrid key={ cat } category={ cat } />
                ))
            }
        </>
    )
}