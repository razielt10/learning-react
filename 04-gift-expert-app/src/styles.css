* {
    font-family: Arial, Helvetica, sans-serif;
    margin: 0;
    box-sizing: border-box;
    margin-block: 0;
}

body {
    margin: 20px;
    background-color: aliceblue;
}

h1 {
    font-size: 30px;
    margin: 10px 0;
}

input {
    padding: 5px;
    border-radius: 5px;
    border: 1px solid rgb(116, 116, 116);
}

button {
    padding: 5px;
    border: none;
    background-color: darkkhaki;
    color: white;
    margin: 0 5px;
    cursor: pointer;
}

button:hover {
    background-color: darkgoldenrod;
}

form.search-category {
    background-color: blanchedalmond;
    margin: 20px 0;
    border-radius: 10px;
    padding: 10px;
}

form span.text-search{
    font-weight: bold;
    display: block;
    font-size: 15px;
}

.category {
    background-color: blanchedalmond;
    padding: 10px;
    border-radius: 10px;
    margin-top: 10px;
}

.category-items {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr 1fr;
}

.category-items img {
    max-width: 100%;
}