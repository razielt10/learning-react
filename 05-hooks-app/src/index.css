* {
    margin: 0;
    padding: 0;
}

body {
    padding: 20px;
}

button {
    margin-right: 10px;
}

a {
    margin: 2px;
    display: inline-block;
 }