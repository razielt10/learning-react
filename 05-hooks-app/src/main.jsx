import React from 'react'
import ReactDOM from 'react-dom/client'
import { HooksApp } from './HooksApp'
import './index.css'
// import { Memorize } from './06-memos/Memorize'
// import { MemoHook } from './06-memos/MemoHook'
import { CallbackCounter } from './06-memos/CallbackCounter'

// import { SimpleForm } from './02-useEffect/SimpleForm'
// import { FormCustomHook } from './02-useEffect/FormCustomHook'
// import { MultipleCustumHooks } from './03-examples/MultipleCustumHooks'
// import { FocusGreen } from './04-useRef/FocusGreen'
// import { Layout } from './05-useLayoutEffect/Layout'
// import { CounterApp } from './01-useState/CounterApp'
// import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook'

ReactDOM.createRoot(document.getElementById('root')).render(
  // <React.StrictMode>
  <>
    <HooksApp />
    {/* <SimpleForm /> */}
    {/* <FormCustomHook /> */}
    {/* <MultipleCustumHooks /> */}
    {/* <FocusGreen /> */}
    {/* <Layout /> */}
    <CallbackCounter />  
  </>
  // </React.StrictMode>,
)
