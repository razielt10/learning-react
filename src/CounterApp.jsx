import { useState } from 'react';
import PropTypes from 'prop-types';

//con fragment
export const CounterApp = ({value}) => {

    const [counter , setCounter] = useState(value)

    function handleAdd (ev) {
        // setCounter(counter + 1)
        setCounter( (c) => c + 1 )
    }   
    function handleSubstract (ev) {
        // setCounter(counter + 1)
        setCounter( (c) => c - 1 )
    }  
    function handleReset (ev) {
        // setCounter(counter + 1)
        setCounter( (c) => value )
    }

    return (
    <>
        <h1>Counter App</h1>
        <h2>{ counter }</h2>
        <button onClick={ handleAdd }>+1</button>
        <button onClick={ handleSubstract }>-1</button>
        <button onClick={ handleReset }>Reset</button>
    </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number
}

CounterApp.defaultProps = {
    value: 0
}
