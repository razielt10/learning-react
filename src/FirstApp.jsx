import { Fragment, ReactComponentElement } from "react"
import PropTypes from 'prop-types';

//con fragment
export const FirstApp = ({title, subTitle, name}) => {

  return (
    <Fragment>
        <h1 data-testid="test-title">{ title }</h1>
        <h2>{ subTitle }</h2>
        <h2>{ subTitle }</h2>
        <h3>{ name }</h3>
    </Fragment>
    
  )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string,
    name: PropTypes.string
}

FirstApp.defaultProps = {
    subTitle: 'Desempleado',
    name: 'Jcgl Juan Carlos'
}

//sin fragment
export const FirstAppSinFragment = () => {
    return (
        <>
            <h4>Yo mando</h4>
            <h5>Ambos sin fragment</h5>
        </>
    )
}