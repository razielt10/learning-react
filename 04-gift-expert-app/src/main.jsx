import React from 'react'
import ReactDOM from 'react-dom/client'
import { GifExperApp } from './GifExpertApp'

import './styles.css'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <GifExperApp />
  </React.StrictMode>,
)
