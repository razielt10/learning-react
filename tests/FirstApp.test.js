import { render } from '@testing-library/react'
import { FirstApp } from '../src/FirstApp'


describe('tests on FirstApp', () => {
  
    test('should be equal an snapshot', () => {
      
        const title = 'Hello!!!'
        const {container } = render( <FirstApp title={title} /> )

        expect(container).toMatchSnapshot()
    });
    

    test('should be should be title (snapshot)', () => {
      
        const title = 'Hello!!!'
        const { getByText, getByTestId } = render( <FirstApp title={title} /> )

        expect( getByText(title) ).toBeTruthy()
        expect( getByTestId('test-title').innerHTML).toContain(title)
    });

    test('should show subtitle send on props', () => {
        const title = 'Hello!!!'
        const subTitle = 'Suntiltulo'
        const { getByText, getAllByText } = render( 
        <FirstApp 
            title={title}
            subTitle={subTitle}
        />
        )

        //hacemos una prueba para un array de elementos con e mismo texto
        expect( getAllByText(subTitle) ).toBeTruthy()
        expect( getAllByText(subTitle).length ).toBe(2)
  
    })
    
    
});
