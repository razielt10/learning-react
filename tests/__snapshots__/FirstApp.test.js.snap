// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on FirstApp should be equal an snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hello!!!
  </h1>
  <h2>
    Desempleado
  </h2>
  <h2>
    Desempleado
  </h2>
  <h3>
    Jcgl Juan Carlos
  </h3>
</div>
`;
